### PR TITLE
Mag. Simulation: fix if statement for self.chi

### PR DESCRIPTION
**Pull request by thast**
When chi is None, `properties` throw an error on the call of `self.chi`, thus the `if` statement ends in an error when true. (see screenshot below)
this is fixed by using the `getattr` function.
Also, this PR fixes the size of model when the chiMap is a `maps.wires`
image

### CODE REVIEW

**Comment by thast** [comment]
An actually, should that not be `getattr(self, 'model', None)` instead? I am having issue with joint inversion because chi and model are of different size and `properties` again is unhappy.

**Review by jcapriot — COMMENTED** [review]
Shouldn't we just do a `self.model = m` in all cases (similar to the JtJ diag routine)?

**Comment by thast** [comment]
@jcapriot, I stayed close to the original formulation (with the `np.zeros`) but I agree with you. It seems to work the same. @fourndo @domfournier might have an opinion or insights?

**Comment by jcapriot** [comment]
I think this assignment is better, as it would properly update the chiDeriv property as well.

**Comment by jcapriot** [comment]
Also, I was suggesting always assigning `self.model = m` without even doing the check for `self.chi`

File: SimPEG/potential_fields/magnetics/simulation.py [hunk 1]
```diff
@@ -196,8 +196,8 @@ def Jtvec(self, m, v, f=None):
     @property
     def fieldDeriv(self):
 
-        if self.chi is None:
-            self.model = np.zeros(self.G.shape[1])
+        if getattr(self, 'chi', None) is None:
+            self.model = m
```
Comment (jcapriot) [inline SimPEG/potential_fields/magnetics/simulation.py]: I would guess the `self.model = m` line would probably not work right here, as we never pass `m` to fieldDeriv (which might not be a good thing).

Comment (thast) [reply SimPEG/potential_fields/magnetics/simulation.py]: I noticed that, it is already been fixed.

Comment (thast) [reply SimPEG/potential_fields/magnetics/simulation.py]: For the same reason, I kept the `if getattr(self, 'chi'` ` for that specific case.

**Pull request merged**